From 0fbde2a07928ba79b7a4ddb4585f6b1de36a74bb Mon Sep 17 00:00:00 2001
From: Thibault Saunier <tsaunier@igalia.com>
Date: Fri, 25 May 2018 12:28:04 +0200
Subject: [PATCH] qtdemux: Clarify field name about stream-encryption-system

This field is actually only informatory and the user can potentially
choose something else. EME tests in WebKit testsuite actually doesn't
take it into and force another encryption system to be used, and expects
to be given the occasion to do so.

This basically also reverts 3e063703b3a51b8aaa7f75f36c4660c583a60e93.
---
 gst/isomp4/qtdemux.c | 13 +++++--------
 1 file changed, 5 insertions(+), 8 deletions(-)

diff --git a/gst/isomp4/qtdemux.c b/gst/isomp4/qtdemux.c
index bd3a00ccf..515687353 100644
--- a/gst/isomp4/qtdemux.c
+++ b/gst/isomp4/qtdemux.c
@@ -8096,12 +8096,6 @@ gst_qtdemux_request_protection_context (GstQTDemux * qtdemux,
   filtered_sys_ids = gst_protection_filter_systems_by_available_decryptors (
       (const gchar **) qtdemux->protection_system_ids->pdata);
 
-  if (!filtered_sys_ids) {
-    GST_INFO_OBJECT (element,
-        "No avalaible decryptor, not worth asking the user to choose.");
-    return;
-  }
-
   g_ptr_array_remove_index (qtdemux->protection_system_ids,
       qtdemux->protection_system_ids->len - 1);
   GST_TRACE_OBJECT (qtdemux, "detected %u protection systems, we have "
@@ -8133,7 +8127,8 @@ gst_qtdemux_request_protection_context (GstQTDemux * qtdemux,
   query = gst_query_new_context ("drm-preferred-decryption-system-id");
   st = gst_query_writable_structure (query);
   gst_structure_set (st, "track-id", G_TYPE_UINT, stream->track_id,
-      "stream-encryption-systems", G_TYPE_STRV, filtered_sys_ids, NULL);
+      "available-stream-encryption-systems", G_TYPE_STRV, filtered_sys_ids,
+      NULL);
   gst_structure_set_value (st, "stream-encryption-events", &event_list);
   if (gst_qtdemux_run_query (element, query, GST_PAD_SRC)) {
     gst_query_parse_context (query, &ctxt);
@@ -8157,7 +8152,9 @@ gst_qtdemux_request_protection_context (GstQTDemux * qtdemux,
         "drm-preferred-decryption-system-id");
     st = (GstStructure *) gst_message_get_structure (msg);
     gst_structure_set (st, "track-id", G_TYPE_UINT, stream->track_id,
-        "stream-encryption-systems", G_TYPE_STRV, filtered_sys_ids, NULL);
+        "available-stream-encryption-systems", G_TYPE_STRV, filtered_sys_ids,
+        NULL);
+
     gst_structure_set_value (st, "stream-encryption-events", &event_list);
     gst_element_post_message (element, msg);
   }
-- 
2.17.0

